python-avd/pyavd/_schema/models/avd_model.py

```diff
@@ -351,6 +393,20 @@ def _cast_as(self, new_type: type[T_AvdModel], ignore_extra_keys: bool = False)
                 raise TypeError(msg)
 
             new_args[field] = value
-            continue
 
-        return new_type(**new_args)
+        new_instance = new_type(**new_args)
+
+        if self._created_from_null:
+            new_instance._created_from_null = True
+        if self._block_inheritance:
+            new_instance._block_inheritance = True
```

Comment:
seems that we don't really need the if checks given that if they are False we set them to False as well

```suggestion
    # Pass along the flags
    new_instance._created_from_null = self._created_from_null
    new_instance._block_inheritance = self._block_inheritance
```